addon/PlacesProvider.js

```diff
@@ -359,6 +359,22 @@ Links.prototype = {
       params.beforeDate = beforeDate;
     }
 
+    // setup filter binding and sql clause
+    let filterClause = "";
+    if (filter) {
+      // Match each token in either the title or url
+      let tokens = filter.trim().toLowerCase().split(/\s+/);
+      tokens.forEach((token, i) => {
+        filterClause = `${filterClause} AND (
```

Comment:
The other Clauses start with "AND ..." because they're chaining to the `WHERE`. This is the same except there's a dynamic number of chains based on how many tokens are being used, e.g., " AND token1 ... AND token2 ... AND token3 ..." The empty string is the initial value so that the subsequent token clauses can be concatenated.